vyper/utils.py

```diff
@@ -166,4 +168,6 @@ def is_varname_valid(varname, custom_units):
         return False
     if varname.lower() in built_in_functions:
         return False
+    if not re.match('^[_a-z][a-zA-Z0-9_]*$', varname.lower()):
```

Comment:
The regex should change to `re.match('^[_a-z][a-z0-9_]*$', varname.lower()` or `re.match('^[_a-zA-Z][_a-zA-Z0-9]*$', varname`. If there's a `.lower()` then there's no need to include `A-Z` in the regex. `A-Z` should be allowed as the first character of the identifier if `.lower()` is not being used.

Both the options look fine to me I recommend dropping `.lower()` and allowing `A-Z` to be the first char.